### PR TITLE
Testing for Classifier

### PR DESCRIPTION
**Pull request by yechenh**

### CODE REVIEW

File: classifier/transformer_decoder_test.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+import numpy as np
+import tensorflow as tf
+import unittest
+from unittest import TestCase
+
+import transformer_decoder
+from official_transformer import model_params
+
+
+class TransformerDecoderTestCase(TestCase):
```
Comment (XiangZeng2020) [inline classifier/transformer_decoder_test.py:26]: s/TestCase/tf.test.TestCase

File: classifier/transformer_decoder_test.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            decoder = transformer_decoder.TransformerDecoder(
+                params=model_params.BASE_PARAMS, train=True)
+            result = decoder.__call__(inputs, encoder_outputs, targets)
+
+            result_dimensions = result.get_shape().as_list()
+            self.assertEqual(result_dimensions, [batch_size, input_length, model_params.BASE_PARAMS['vocab_size']])
+        
+
+if __name__ == '__main__':
+    unittest.main()
```
Comment (XiangZeng2020) [inline classifier/transformer_decoder_test.py:65]: s/unittest.main()/tf.test.main()

File: classifier/transformer_decoder_test.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        target_length = 128
+        
+        tf.reset_default_graph()
+        with tf.Session() as sess:
+            inputs = tf.constant(np.zeros([batch_size, input_length]), dtype="float32")
+            encoder_outputs = tf.constant(np.zeros([batch_size, input_length, hidden_size]), dtype="float32")
+            targets = tf.constant(np.zeros([batch_size, target_length]), dtype="int32")
+            decoder = transformer_decoder.TransformerDecoder(
+                params=model_params.BASE_PARAMS, train=True)
+            result = decoder.__call__(inputs, encoder_outputs, targets)
```
Comment (XiangZeng2020) [inline classifier/transformer_decoder_test.py:58]: just 

decoder(inputs, encoder_outputs, targets)

File: classifier/transformer_decoder_test.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        hidden_size = 768
+        
+        tf.reset_default_graph()
+        with tf.Session() as sess:
+            inputs = tf.constant(np.zeros([batch_size, input_length]), dtype="float32")
+            encoder_outputs = tf.constant(np.zeros([batch_size, input_length, hidden_size]), dtype="float32")
+
+            decoder = transformer_decoder.TransformerDecoder(
+                params=model_params.BASE_PARAMS, train=True)
+            result = decoder.__call__(inputs, encoder_outputs)
```
Comment (XiangZeng2020) [inline classifier/transformer_decoder_test.py:40]: result = decoder(inputs, encoder_outputs)

File: classifier/transformer_decoder_test.py (hunk 1 above)
Comment (yechenh) [reply classifier/transformer_decoder_test.py]: Done

File: classifier/transformer_decoder_test.py (hunk 2 above)
Comment (yechenh) [reply classifier/transformer_decoder_test.py]: Done

File: classifier/transformer_decoder_test.py (hunk 3 above)
Comment (yechenh) [reply classifier/transformer_decoder_test.py]: Done

File: classifier/transformer_decoder_test.py (hunk 4 above)
Comment (yechenh) [reply classifier/transformer_decoder_test.py]: Done

**Pull request merged**